wandb/sdk/wandb_run.py

```diff
@@ -2950,6 +2950,10 @@ def link_artifact(
         if self._backend and self._backend.interface:
             if artifact.is_draft() and not artifact._is_draft_save_started():
                 artifact = self._log_artifact(artifact)
+
+            # Wait until the artifact is committed before trying to link it.
+            artifact.wait()
```

Comment:
Done! I was trying to minimize changes here, but rearranging the code makes it clearer. Let me know what you think!

The `artifact.wait()` now happens after the offline-mode check.

Reply:
Yeah, much better. Since this is feedback you've given me I thought you wouldn't mind it, though I appreciate the tension of trying to keep the PR small